src/triage/experiments/base.py

```diff
@@ -595,6 +604,17 @@ def _run(self):
 
         self.train_and_test_models()
         logging.info("Experiment complete")
+        if self.profile:
+            cp.disable()
+            store = self.project_storage.get_store(
+                ["profiling_stats"],
+                f"{int(time.time())}.profile"
+            )
+            with store.local_fs_path() as fname:
+                cp.dump_stats(fname)
+                logging.info("Profiling stats of this Triage run calculated and written to %s"
+                             "in cProfile format.",
+                             store)
```

Comment:
I'm sympathetic to the desire to use the `dump_stats` method; however, especially considering the extra `Store` code that had to be added to support it, (and considering the _potential_ performance implications for the `S3Store`), I seriously wonder if it might be better to do it ourselves, (using methods that are no less "public"):

```
import marshal

…

def _run_profile(self):
    cp = cProfile.Profile()
    cp.runcall(self._run)
    store = …
    with store.open('wb') as fd:
        cp.create_stats()
        marshal.dump(cp.stats, fd)
    logging.info(…)
```

(See `cProfile.Profile.dump_stats??` in iPython.)

Reply:
ah this is interesting. I should change it over to this